doc/anjuta-launcher.1.in
.TH "anjuta-launcher" "1" "@VERSION@" "Jens Georg" ""
.SH "NAME"
.LP
anjuta-launcher \- GNOME Integrated Development Environment
.LP
anjuta-launcher normally is not called directly by the user but invoked by
anjuta
.SH "SYNTAX"
.LP
anjuta-launcher [ \fIoption\fP ] program_name [ \fIprogram_parameter ... \fP]
.SH "DESCRIPTION"
This program is part of the Anjuta GNOME IDE. It is used to launch programs
from within the IDE and to initialize a terminal for a debugging session.
.SH "OPTIONS"
.TP
\fB\-\-version\fR
Output version information and exit.
.TP
\fB\-\-__debug_terminal\fR
Initalize a terminal for a debugging session. With this options, the second
parameter is a temporary file where the name of the tty/terminal is written
to.
.SH "AUTHORS"
.LP 
This man page was written by Jens Georg <mail@jensgeorg.de>. Anjuta was developed by Kh. Naba Kumar Singh <kh_naba@users.sourceforge.net>
.SH "SEE ALSO"
.LP 
anjuta(1)


NEWS
Anjuta 2.28.1 (19 October 2009) -- Johannes Schmid

Bugs fixed:
Fix check if the executable is part of the debug configuration
bgo#597318 -  Anjuta terminal crashes X
symbol-db: fix #597113
subversion: Fix duplicate IDs in the glade file (bgo 596001)
file-loader: bgo#567363 -  drag and drop from gnome-search-tool yields garbage message

Anjuta 2.28.0 (28 September 2009) -- Johannes Schmid

Bugs fixed:
gdb: bgo#594665 -  Fix link of test program
sourceview: Report error when location is > 0 for markers (bgo#593954)
terminal: bgo#594648 -  Terminal does not work
sourceview: #355151 -  drag/drop into editor window should open file

And of course lots of translation updates!

Anjuta is translated fully into the following languages:
pt_BR, en_GB, bg, de, da, fr, gl, it, pt, sv, es, tr, uk, hu, th

The manual is fully translated for:
el, es and th

Anjuta 2.27.92 (7 September 2009) -- Johannes Schmid

Bugs fixed:
class-gen: bgo #593532 - Untranslated strings
language-support-cpp-java: bgo #593291 - fix small typo

Thanks to all translators!

Anjuta 2.27.91 (27 August 2009) -- Johannes Schmid

Updates:
- Some git fixes (file-status, etc.)
- Stability fixes on symbol-db

Bugs fixed:
Typo in Anjuta pot file (bgo#593236)
bgo #592941 – Fixed errors in Anjuta IDE Manual (Greek translation)
bgo #591758 -  Fix language mistakes
bgo #591887 -  Bad strings for translation
bgo #590117 – Build failure: local scandir() function collides with system's
bgo #589828 – Can't open GtkBuilder files with the glade plugin
bgo#590708 – Some untranslated strings
project-wizard: bgo#591359 – Need an easier and intuitive way to create a shared library project
bgo #590337 – Compile failure: not allowed to link against a -module
bgo #590720 – glade plugin FTBFS on mips(el) and alpha, LANGUAGE_C built-in
bgo#590337 – Compile failure: not allowed to link against a -module
language-support: bgo#575926 – 'Maintain past indentation' disables itself
bgo #578690 – Several components underlinked
bgo #585428 – Context needed to translate
bgo#590334 – Fix -I ordering
bgo #590338 – Missing plugins flags
git: Add init support (bgo 566698)
language-support-cpp-java: fixed critical #588112 - infinite loop / freeze
bgo#587248 – Encoding key in desktop file is deprecated
bgo#588485 – fixes from running anjuta with G_DEBUG="fatal_warnings"
bgo#589699 – When ~/.cache isn't exists, session data isn't saved
bgo#588506 – Wrong value of PACKAGE_DOC_DIR
bgo#589601 – "gtk-execute" and "gtk-cancel" marked for translation

Anjuta 2.27.5 (27 July 2009) -- Johannes Schmid

Updates:
- Even better git plugin (James Liggett)
- Work to get C++ autocompletion done (Massimo' Cora)

Bugs fixed:
git: Show the active branch in the status bar (bgo 589401)
symbol-db: #588387 make typeahead searching caseless
Migrate anjuta project wizards from libglade to GtkBuilder (GnomeBug:530740)
bgo #575147 – Headers of new C++/GObject classes have wrong copyright and license notices.
bgo #577741 – "single step over function" does not always work as expected
bgo #587957 – doap file missing bug-database
bgo #583731 – Some untranslated strings
git: Refresh the log view when switching branches (bgo 580276)
bgo#578438 – broken svn libraries detection in configure
git: Fix a typo in a menu tooltip (bgo 587191)
gbf-am: build fix #586926
#586926 – gdbmi_test build fails
bgo#513156 – Get rid of libgnome(ui)
git: Add descriptions to the menu items (bgo 577924)
#586621 – Crash while loading project file
go#586148 – Case had to be the other way around
bgo#586148 – Fixing setting whitespace for the editor not working
bgo#577924 – Fixing AnjutaStatusbar for displaying menu tags
#577232 – run-program plugin: projects's working directory should be set to project root on start
#586030 – Value of a new environment variable is not saved in run parameter dialog
#584180 – Duplicate copy of GBF/General.pm installed to same location
#582961 – Replace SexyIconEntry by GtkEntry icon functionality (available in GTK+ 2.16)
subversion: Show conflicted files in the commit dialog (bgo 578411)
symbol-db: #585498 – Completion stop working and strange warning

Anjuta 2.27.3 (14 June 2009) -- Johannes Schmid

Updates:
- Pixel-saving UI for tabs
- Remove markup from translatable strings

Bugs fixed:
- Lots of git bugs fixed
#584013 – libanjuta-1.0.pc.in still requires libglade-2.0
#584347 – Don't reference child revisions
#567214 - bookmarks window wastes too much space
#582464 – crash in Anjuta IDE: Hacking on Empathy, dele...

Anjuta 2.27.2 (26 May 2009) -- Johannes Schmid

Updates:
- Lots of git improvements
- Subversion improvements
- Removed libglade dependency

#581416 - Fix void function cannot return value
#581869 - anjuta fails to build outside source tree
#582710 - Please consider using g_timeout_add_seconds to reduce wakeups
#582977 – String misspelled
#581421 - build fail on symbol-db/test
#530740 – Use GtkBuilder instead of libglade
#583589 – Some untranslated strings
#580340 - Reuse the last commit message when amending commits

Anjuta 2.27.1 (4 May 2009) -- Johannes Schmid

Updates:
- Improvements in git plugin
- Improvements with auto-completion speed

Bugs fixed:
#530740 - Use GtkBuilder instead of libglade
#581074 - Creating a new wxWidgets project should not include libglade
#580994 - using DEBUG_PRINT instead of g_print
#580013 – patch plugin doesn't handle patch-files with whitespace
#568779 – project wizard should infer project name from base path
#577721 – tooltips appears in wrong position
#579118 – Segmentation fault when adding file to project
#578087 – Artwork for Symbol Database is actually not from Monodevelop
#567588 – highlight default button in Find dialog
#576959 - Search does not point to correct line number on first match when file is not opened in editor.
#577006 – crash creating new file
#574607 – Replace function selects text uncorrectly
#576057 – doesn't build with graphviz 2.22.2
#575617 – Update FSF address

Anjuta 2.26.0 (16 March 2009) -- Johannes Schmid


launcher/anjuta-launcher.c
/*
    anjuta-launcher.c
    Copyright (C) 2000  Kh. Naba Kumar Singh

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
*/

#ifdef HAVE_CONFIG_H
#  include <config.h>
#endif

#include <sys/types.h>
#include <sys/unistd.h>
#include <sys/wait.h>
#include <stdio.h>
#include <glib.h>
#include <unistd.h>
#include <string.h>
#include <stdlib.h>

/* #define DEBUG */

int
main (int argc, char **argv)
{
	pid_t pid;
	int status, i;
	char **arg_v;

	if (argc < 2)
	{
		printf ("\nUsage:\n");
		printf
			("anjuta-launcher program_name [program_parameter ... ]\n\n");
		exit (-1);
	}

	if (strcmp (argv[1], "--version") == 0)
	{
		printf ("anjuta-launcher version 0.1.2\n");
		exit (0);
	}
	if (strcmp (argv[1], "--__debug_terminal") == 0)
	{
		FILE *fp;
		
		if (argc != 3)
		{
			printf ("\nUsage:\n");
			printf ("anjuta-launcher program_name [program_parameter ... ]\n\n");
			exit (-1);
		}
		fp = fopen (argv[2], "w");
		if (fp == NULL)
		{
			g_warning ("Fatal Error: Cannot write to redirection file\n");
		} 
		else
		{
			char* tty_name;
			tty_name = ttyname(0);
			if (tty_name)
			{
				fprintf (fp, "%s\n", tty_name);
				fclose (fp);
				printf ("Debug Terminal for the process:\n");
				printf ("-------------------------------\n");
				while (1) pause ();
				exit (0);
			}
			else
			{
				fprintf (fp, "__ERROR__\n");
				fclose (fp);
				g_warning ("Fatal Error: Unexpected error -- Unknowm tty\n");
			}
		}
		getchar ();
		exit (-1);
	}

	arg_v = g_malloc ((argc) * sizeof (char *));
	printf ("EXECUTING:\n");
	for (i = 1; i < argc; i++)
	{
		printf ("%s ", argv[i]);
		arg_v[i - 1] = g_strdup (argv[i]);
	}
	printf ("\n----------------------------------------------\n");
	arg_v[i - 1] = NULL;

	if ((pid = fork ()) == 0)
	{
		execvp (argv[1], arg_v);
		g_error ("Unable to execute the command (not found)\n");
	}
	if (pid < 0)
	{
		printf ("\n----------------------------------------------\n");
		printf ("There was an error in launching the program\n");
		status = -1;
	}
	else
	{
		waitpid (pid, &status, 0);
		printf ("\n----------------------------------------------\n");

		waitpid (pid, &status, 0);
		
		if (WIFSIGNALED (status)) {
			int signal = WTERMSIG (status);

			printf ("Program has been terminated receiving signal %d (%s)\n", signal, g_strsignal (signal));
		} else if (WIFEXITED (status))
			printf ("Program exited successfully with errcode (%d)\n", WEXITSTATUS (status));
	}
	printf ("Press the Enter key to close this terminal ... \n");
	getchar ();
	return WEXITSTATUS (status);
}
